moveit_core/collision_detection_bullet/test/test_bt_collision_detection_panda.cpp

```diff
@@ -0,0 +1,354 @@
+  ROS_INFO_STREAM("Continous to continous collisions are not supported yet, therefore fail here.");
+  ASSERT_TRUE(res.collision);
+  res.clear();
+}
+
+/** \brief Two similar robot poses are used as start and end pose of a continuous collision check. */
+TEST_F(BulletCollisionDetectionTester, ContinuousCollisionWorld)
+{
+  collision_detection::CollisionRequest req;
+  req.contacts = true;
+  req.max_contacts = 10;
+  collision_detection::CollisionResult res;
+
+  robot_state::RobotState state1(robot_model_);
+  robot_state::RobotState state2(robot_model_);
+
+  setToHome(state1);
+  state1.update();
+
+  setToHome(state2);
+  double joint_2{ 0.05 };
+  double joint_4{ -1.6 };
+  state2.setJointPositions("panda_joint2", &joint_2);
+  state2.setJointPositions("panda_joint4", &joint_4);
+  state2.update();
+
+  cworld_->checkRobotCollision(req, res, *crobot_, state1, state2, *acm_);
+  ASSERT_FALSE(res.collision);
+  res.clear();
+
+  // Adding the box which is not in collision with the individual states but with the casted one.
+  shapes::Shape* shape = new shapes::Box(0.1, 0.1, 0.1);
+  shapes::ShapeConstPtr shape_ptr(shape);
+
+  Eigen::Isometry3d pos{ Eigen::Isometry3d::Identity() };
+  pos.translation().x() = 0.43;
+  pos.translation().y() = 0;
+  pos.translation().z() = 0.55;
+  cworld_->getWorld()->addToObject("box", shape_ptr, pos);
+
+  cworld_->checkRobotCollision(req, res, *crobot_, state1, *acm_);
+  ASSERT_FALSE(res.collision);
+  res.clear();
+
+  cworld_->checkRobotCollision(req, res, *crobot_, state2, *acm_);
+  ASSERT_FALSE(res.collision);
+  res.clear();
+
+  cworld_->checkRobotCollision(req, res, *crobot_, state1, state2, *acm_);
+  ASSERT_TRUE(res.collision);
+  ASSERT_EQ(res.contact_count, 4u);
+  res.clear();
+}
+
+/** \brief Tests the padding through expanding the link geometry in such a way that a collision occurs. */
+TEST_F(BulletCollisionDetectionTester, PaddingTest)
+{
+  collision_detection::CollisionRequest req;
+  req.contacts = true;
+  req.max_contacts = 10;
+  collision_detection::CollisionResult res;
+
+  cworld_->checkRobotCollision(req, res, *crobot_, *robot_state_, *acm_);
+  ASSERT_FALSE(res.collision);
+  res.clear();
+
+  // Adding the box right in front of the robot hand
+  shapes::Shape* shape = new shapes::Box(0.1, 0.1, 0.1);
+  shapes::ShapeConstPtr shape_ptr(shape);
+
+  Eigen::Isometry3d pos{ Eigen::Isometry3d::Identity() };
+  pos.translation().x() = 0.43;
+  pos.translation().y() = 0;
+  pos.translation().z() = 0.55;
+  cworld_->getWorld()->addToObject("box", shape_ptr, pos);
+
+  crobot_->setLinkPadding("panda_hand", 0.08);
+  cworld_->checkRobotCollision(req, res, *crobot_, *robot_state_, *acm_);
+  ASSERT_TRUE(res.collision);
+  res.clear();
+
+  crobot_->setLinkPadding("panda_hand", 0.0);
+  cworld_->checkRobotCollision(req, res, *crobot_, *robot_state_, *acm_);
+  ASSERT_FALSE(res.collision);
+}
+
+int main(int argc, char** argv)
+{
+  testing::InitGoogleTest(&argc, argv);
```

Comment:
great tests!

Reply:
Fully agree. Can we make these tests generic as well, i.e. to be used for both FCL and Bullet?

Reply:
Yes, done.